### ISSUE TITLE
Enhance Array Type and String Type

### ISSUE THREAD

**Issue by llmhyy** [sunjun-group/Ziyuan#149]
hi @lylytran 

In general, we would like the input length to be "fixed" so that each input can be regarded as a "vector" in space. In general, the data structure like array, list, queue, are flexible and dynamic. Therefore, the idea is to pre-assign a space in vector for them. Assume there is a data structure `Student` as follows:  
```
class Student{
  int id,
  String name,
  int[] scores,
  Student bestFriend,
  Student[] friends
}
```

We allocate a fixed space `m` for each complex data type such as `Student`, `String`, and `int[]`. When vectorize a complex data type, we (1) calculate its required slot in a top-down manner, and (2) filling information into those slots.

Take `Student` for example, given a predefined size as `k`, we assign different number of slots for its fields. 
- Primitive type: one slot
- Array type: `k` slots, the first slot indicates null or not, the second slot indicates length, the rest `k-2` slots indicate content. If the first slot is 0, the rest `k-1` elements are considered as **padding elements**. If the content length `cl` is less than `k`, the rest `k-2-cl` elements are considered as padding elements. In addition, for multiple dimension array like `int[][][]`, we assign `k*k*k` slots.
- String type: we can treat it as char[].
- Object type: `1+m` slots, the first slot indicates whether it is null, the following `m` slots indicates the cumulatively calculated length of its fields. The calculate `m` as follows. If a field is a complex type, all its complex-type fields are null. For example, the Student has a `bestFriend` of `Student` type, the `bestFridend` field takes 1 (null or not) + 1 (id) + k (name) + k (scores) + 1 (bestFriend null or not, but it must be null, i.e., not modifiable) + 1 (friends null or not, but it must be null, i.e., not modifiable)  = 2k+3 slots. Note that, if the first element is 0, all the following elements are padding elements. In short, Object type can expand at most 2-layers. Of course, we may modify the above formula if we allow 3 or more layers.

In summary,  given a complex type, we can calculate its fixed length. In the example, we assign 1+k+k+(2k+3)+k*(2k+3) slots. Note that, now each `ExecType` should describe whether it is a padding element, a modifiable element, or a normal element, as well as parent-child relation if necessary. Those information is important to transform a vector back to a test case. 

Thanks!
